### PR TITLE
Enable ginkgolinter

### PR DESCRIPTION
**Pull request by bshewale**

### CODE REVIEW

File: api/bases/designate.openstack.org_designateapis.yaml
[the remark this answers is not in the document]
Comment (bshewale) [reply api/bases/designate.openstack.org_designateapis.yaml:1]: it was automatically added by running "pre-commit run -a" command.

**Review by beagles — APPROVED** [review]

File: api/v1beta1/designatecentral_types.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ type DesignateCentralSpec struct {
 
 	// +kubebuilder:validation:Required
 	// Secret containing OpenStack password information for designate DesignateDatabasePassword, AdminPassword
-	Secret string `json:"secret,omitempty"`
```
Comment (marios) [inline api/v1beta1/designatecentral_types.go:57]: so, ginkgolinter does not like the omitempty tag?

File: api/bases/designate.openstack.org_designateagents.yaml [hunk 2]
```diff
@@ -0,0 +1,258 @@
+---
```
Comment (marios) [inline api/bases/designate.openstack.org_designateagents.yaml:1]: what is the story with all of these new files.
is that a ginkgolinter thing?

**Review by marios — COMMENTED** [review]
i really don't know anything about ginkgolinter, so i took the opportunity to dig for 10 mins.

adding some links here in case useful for anyone else:


https://golangci-lint.run/usage/linters/
https://onsi.github.io/ginkgo/#top
https://github.com/nunnatsa/ginkgolinter


my main concern currently is do we understand why these new files are being created? If we don't then we might reconsider the use of ginkgolinter until we do.

File: api/v1beta1/designatecentral_types.go (hunk 1 above)
Comment (abays) [reply api/v1beta1/designatecentral_types.go:57]: I think our `operator-lint` linter should be flagging this.  Perhaps we don't have that linter here though.  Anyhow, `Required` and `omitempty` are mutually exclusive, as `omitempty` is basically synonymous with `Optional`.

File: api/bases/designate.openstack.org_designateagents.yaml (hunk 2 above)
Comment (abays) [reply api/bases/designate.openstack.org_designateagents.yaml:1]: These are created by `make manifests`: https://github.com/openstack-k8s-operators/designate-operator/blob/5e2c14161b3f1aff5beca3a15fe2293c47d51401/Makefile#L91

Comment (bshewale) [reply api/bases/designate.openstack.org_designateagents.yaml:1]: > what is the story with all of these new files. is that a ginkgolinter thing?

Hey Marios, "pre-commit run -a" command added all the changes.

File: api/v1beta1/designatecentral_types.go (hunk 1 above)
Comment (bshewale) [reply api/v1beta1/designatecentral_types.go:57]: Yeah make-operator-lint failed with below error:

make-operator-lint.......................................................Failed
- hook id: make-operator-lint
- exit code: 2

test -f go.work || go work init
go work use .
go work use ./api
go work sync
GOBIN=/home/bhagyashris/src/next_gen/operators/designate-operator/bin go install github.com/gibizer/operator-lint@v0.1.0
go vet -vettool=/home/bhagyashris/src/next_gen/operators/designate-operator/bin/operator-lint ./... ./api/...
# github.com/openstack-k8s-operators/cinder-operator/api/v1beta1
api/v1beta1/designateagent_types.go:57:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
api/v1beta1/designatecentral_types.go:57:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
api/v1beta1/designatemdns_types.go:60:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
api/v1beta1/designateproducer_types.go:60:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
api/v1beta1/designatesink_types.go:60:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
api/v1beta1/designateworker_types.go:60:2: C001: Field 'Secret' has both a 'Required' kubebuilder marker and an 'omitempty' tag. Only one of them should be used
make: *** [Makefile:300: operator-lint] Error 2

and to resolve that i have removed omitempty.

**Comment by bshewale** [comment]
@abays Hey, can you help here to resolve the "Golang lint, vet and unit test pipeline / github (govet, golint and gotest) (pull_request) " failure it's failing with below error log:

Error: ../../../go/pkg/mod/k8s.io/kube-openapi@v0.0.0-20230308215209-15aac26d736a/pkg/cached/cache.go:242:16: undefined: atomic.Pointer
Error: Process completed with exit code 2.

https://github.com/openstack-k8s-operators/designate-operator/actions/runs/4894241267/jobs/8738726088?pr=40

**Comment by abays** [comment]
> @abays Hey, can you help here to resolve the "Golang lint, vet and unit test pipeline / github (govet, golint and gotest) (pull_request) " failure it's failing with below error log:
> 
> Error: ../../../go/pkg/mod/k8s.io/[kube-openapi@v0.0.0-20230308215209-15aac26d736a](mailto:kube-openapi@v0.0.0-20230308215209-15aac26d736a)/pkg/cached/cache.go:242:16: undefined: atomic.Pointer Error: Process completed with exit code 2.
> 
> https://github.com/openstack-k8s-operators/designate-operator/actions/runs/4894241267/jobs/8738726088?pr=40

I believe `atomic.Pointer` is something added in Golang 1.19.  So we probably need to update to that here [1] and perhaps elsewhere.

[1] https://github.com/openstack-k8s-operators/designate-operator/blob/main/.github/workflows/golangci-lint.yaml

File: pkg/designate/common.go
Comment (bshewale) [inline pkg/designate/common.go:1]: following this https://github.com/golang/lint/issues/488 to resolve the "(govet, golint and gotest)"  job

**Comment by bshewale** [comment]
> > @abays Hey, can you help here to resolve the "Golang lint, vet and unit test pipeline / github (govet, golint and gotest) (pull_request) " failure it's failing with below error log:
> > Error: ../../../go/pkg/mod/k8s.io/[kube-openapi@v0.0.0-20230308215209-15aac26d736a](mailto:kube-openapi@v0.0.0-20230308215209-15aac26d736a)/pkg/cached/cache.go:242:16: undefined: atomic.Pointer Error: Process completed with exit code 2.
> > https://github.com/openstack-k8s-operators/designate-operator/actions/runs/4894241267/jobs/8738726088?pr=40
> 
> I believe `atomic.Pointer` is something added in Golang 1.19. So we probably need to update to that here [1] and perhaps elsewhere.
> 
> [1] https://github.com/openstack-k8s-operators/designate-operator/blob/main/.github/workflows/golangci-lint.yaml

thanks @abays :) now the job is passing after doing some change. Please review when you get time because this PR will unblock most of the PR which are under review.

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**